### PR TITLE
Remove Diaspora implicit self mention

### LINKED ISSUE #7629: Get rid of implicit self-mentions
I'm happy with the implicit mentions we are doing. But I don't like one thing – implicit self-mentions.

### Expected behavior
When I comment my own posts Friendica should not add an implicit self-mention. 

### Actual behavior
When I comment my own posts I automatically mention myself. This seems completely unnecessary and very weird. I my eyes this gives the impression that I'm a self-centered and lunatic person. 😀

### Steps to reproduce the problem
Write a post. Write a comment on the top level post. Friendica doesn't show that self-mention. But on other networks you can see it.

I think implicit self-mentions have no purpose and no practical advantage at all. Please add a check for implicit self-mentions and then suppress (get rid of) them completely.

### Friendica version you encountered the problem
2019.09-rc

### Friendica source (git, zip)
git

### PHP version
7.2

### SQL version
10.1 MariaDB

### PR DESCRIPTION
**Pull request by MrPetovan**
Fixes #7629 

I gave it a quick shot as a favor to @AlfredSK , I didn't feel like testing it because it involves me opening my Diaspora account I registered some pod I'm not sure where.

Please approve and merge with caution.

### CODE REVIEW

**Comment by nupplaphil** [comment]
LGTM, but I'd say in case of untested Protocoll stuff, @annando should check it too ;)

**Comment by AlfredSK** [comment]
I could change the file locally on my node and test it if that's helpful.

File: src/Protocol/Diaspora.php [hunk 1]
```diff
@@ -3850,20 +3850,26 @@ private static function constructComment(array $item, array $owner)
...
 
 		$body = $item["body"];
 
-		if ((empty($item['uid']) || !Feature::isEnabled($item['uid'], 'explicit_mentions'))
+		// The replied to autor mention is prepended for clarity if:
+		// - Item replied isn't yours
+		// - Item is public or explicit mentions are disabled
+		// - Implicit mentions are enabled
+		if (
+			$owner['id'] != $thread_parent_item['author-id']
```
Comment (annando) [inline src/Protocol/Diaspora.php]: The author-id is the value for the public contact. This differs from the id in the owner array. The item should have the author-id as well, so you could compare these two.

Comment (MrPetovan) [reply src/Protocol/Diaspora.php]: Got it, thanks.

**Comment by MrPetovan** [comment]
> I could change the file locally on my node and test it if that's helpful.

You can actually check outmy own branch on your server:
```
$ git remote add MrPetovan https://github.com/MrPetovan/friendica
$ git fetch --all
$ git checkout MrPetovan/bug/7629-diaspora-implicit-self-mention
```
Just do a pull every time I add a commit.

**Comment by AlfredSK** [comment]
...or I git pull the RC. :-)
(I wasn't fast enough)

**Comment by AlfredSK** [comment]
Works as expected! Thank you very much. I am more and more convinced that will be the best Friendica ever! :-)

**Pull request merged**